File: Projeto Didio Pizza/Didio Pizza Node/backend/src/controller/lojaController.js
import { Router } from 'express';
import * as lojaService from '../service/lojaService.js';
import { getAuthentication, onlyAdmin } from '../utils/jwt.js';

const endpoints = Router();
const auth = getAuthentication();
const adm = onlyAdmin();


endpoints.get('/loja/status', async (req, resp) => {
    try {
        let status = await lojaService.verificarStatusLoja();
        resp.status(200).send(status);
    } catch (err) {
        resp.status(400).send({
            erro: err.message
        });
    }
});


endpoints.put('/admin/loja/abrir', auth, adm, async (req, resp) => {
    try {
        await lojaService.abrirLoja();
        resp.send({ 
            msg: 'Loja aberta com sucesso' 
        });
    } catch (err) {
        resp.status(400).send({ 
            erro: err.message 
        });
    }
});


endpoints.put('/admin/loja/fechar', auth, adm, async (req, resp) => {
    try {

        await lojaService.fecharLoja();
        
        resp.send({ 
            msg: 'Loja fechada com sucesso' 
        });
    } catch (err) {
        resp.status(400).send({ erro: err.message });
    }
});

export default endpoints;

File: Projeto Didio Pizza/Didio Pizza Springboot/backend/src/main/java/com/pizzaria/didiopizza/controllers/LojaController.java
package com.pizzaria.didiopizza.controllers;

import java.util.Map;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RestController;

import com.pizzaria.didiopizza.security.JwtUtil;
import com.pizzaria.didiopizza.service.LojaService;

@RestController
public class LojaController {
    private final LojaService service;
    private final JwtUtil jwt;

    public LojaController(LojaService s, JwtUtil j) {
        service = s;
        jwt = j;
    }

    @GetMapping("/loja/status")
    public Map<String, Object> status() {
        return service.verificarStatusLoja();
    }

    @PutMapping("/admin/loja/abrir")
    public Map<String, String> abrir(
            @RequestHeader("Authorization") String a) {
        jwt.onlyAdmin(a);
        service.abrirLoja();
        return Map.of("msg", "Loja aberta com sucesso");
    }

    @PutMapping("/admin/loja/fechar")
    public Map<String, String> fechar(
            @RequestHeader("Authorization") String a) {
        jwt.onlyAdmin(a);
        service.fecharLoja();
        return Map.of("msg", "Loja fechada com sucesso");
    }
}
